### PR TITLE
hiddify-next: init at 1.1.1

### LINKED ISSUE #285375: Package request: hiddify-next
**Project description**

A multi-platform proxy client based on Sing-box universal proxy tool-chain. Hiddify Next offers a wide range of capabilities, like automatic node selection, TUN mode, remote profiles etc. Hiddify Next is ad-free and open-source. With support for a wide range of protocols, it provides a secure and private way for accessing free internet.

Note, From the [Archwiki](https://wiki.archlinux.org/title/V2Ray): V2ray has become mostly "unmaintainable" and has to be replaced by Sing-Box/shadowsocks-go...

**Metadata**

* homepage URL: https://hiddify.com/
* source URL:  https://github.com/hiddify/hiddify-next
* license: Attribution-NonCommercial-ShareAlike 4.0 InternationalJ
* platforms: Windows, Mac, Linux, Android

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by wenjinnn**
init with appimage version
fix #285375, related #287770
also add wenjinnn to maintainer-list
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/hiddify-next/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'LD_LIBRARY_PATH=usr/lib ' ''''''
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Multi-platform auto-proxy client (appimage version)";
+    longDescription = ''
+      Multi-platform auto-proxy client, supporting Sing-box, X-ray, TUIC, Hysteria, Reality, Trojan, SSH etc. It’s an open-source, secure and ad-free.
```
Comment (linsui) [inline pkgs/applications/networking/hiddify-next/default.nix:45]: cc-by-nc-sa-40 is not an open source license. It's not approved by OSI not FSF.

Comment (wenjinnn) [reply pkgs/applications/networking/hiddify-next/default.nix:45]: you‘re right, I've simply copy the repo description. I'll fix it.

**Review by donovanglover — COMMENTED** [review]
Welcome to nixpkgs! Make sure to put new packages in [`pkgs/by-name`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name). You can also format your code with [`nixfmt-rfc-style`](https://search.nixos.org/packages?channel=unstable&show=nixfmt-rfc-style)

**Comment by donovanglover** [comment]
When you're done, make sure that you have 2 separate commits in this order:

1. maintainers: add wenjinnn
2. hiddify-next: init at 1.1.1

File: pkgs/applications/networking/hiddify-next/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
+, appimageTools
+, fetchurl
+}: let
+  pname = "hiddify";
```
[the remark this answers is not in the document]
Comment (wenjinnn) [reply pkgs/applications/networking/hiddify-next/default.nix:6]: Sure! that's more reasonable.

**Comment by wenjinnn** [comment]
> When you're done, make sure that you have 2 separate commits in this order:
> 
> 1. maintainers: add wenjinnn
> 2. hiddify-next: init at 1.1.1

All done! thanks for the suggestion and guide.

File: pkgs/by-name/hi/hiddify-next/package.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  meta = with lib; {
+    description = "Multi-platform auto-proxy client (appimage version)";
+    longDescription = ''
+      Multi-platform auto-proxy client, supporting Sing-box, X-ray, TUIC, Hysteria, Reality, Trojan, SSH etc.
+    '';
+    homepage = "https://github.com/hiddify/hiddify-next";
+    license = licenses.cc-by-nc-sa-40;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ wenjinnn ];
+    mainProgram = "hiddify";
```
Comment (Aleksanaa) [inline pkgs/by-name/hi/hiddify-next/package.nix:48]:
```suggestion
    mainProgram = "hiddify";
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

File: pkgs/by-name/hi/hiddify-next/package.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    url = "https://github.com/hiddify/hiddify-next/releases/download/v${version}/Hiddify-Linux-x64.AppImage";
+    hash = "sha256-T4BWxhJ7q13KE1rvvFsnXhs2XVEmNkFTJbJ4e8PCg+0=";
+  };
+  hiddify = appimageTools.wrapType2 {
+    inherit pname version src;
+
+    extraPkgs = pkgs: with pkgs; [ libepoxy ];
+  };
+in
+stdenv.mkDerivation rec {
```
Comment (Aleksanaa) [inline pkgs/by-name/hi/hiddify-next/package.nix:21]: Not needed to open another derivation. See example: https://github.com/NixOS/nixpkgs/blob/efcd11b433edce0b444180e23bcd66f964050dd9/pkgs/by-name/ja/jan/package.nix

File: pkgs/by-name/hi/hiddify-next/package.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    hash = "sha256-T4BWxhJ7q13KE1rvvFsnXhs2XVEmNkFTJbJ4e8PCg+0=";
+  };
+  appimageContents = appimageTools.extractType2 { inherit pname version src; };
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+
+  extraPkgs = pkgs: with pkgs; [ libepoxy ];
+  extraInstallCommands = ''
+    runHook preInstall
```
Comment (Aleksanaa) [inline pkgs/by-name/hi/hiddify-next/package.nix:21]: Seems `runHook` is not needed.

**Comment by wenjinnn** [comment]
Seems like after 24.05 release, `appimageTools.wrapType2` behavior changed. after 24.05, this command will fail:
```
    mv $out/bin/hiddify-next-1.1.1 $out/bin/hiddify
```
now the location `$out/bin/hiddify-next-1.1.1` change to `$out/bin/hiddify-next`, so I adapted this behavior.

File: pkgs/by-name/hi/hiddify-next/package.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    hash = "sha256-T4BWxhJ7q13KE1rvvFsnXhs2XVEmNkFTJbJ4e8PCg+0=";
+  };
+  appimageContents = appimageTools.extractType2 { inherit pname version src; };
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+
+  extraPkgs = pkgs: with pkgs; [ libepoxy ];
+  extraInstallCommands = ''
+    mv $out/bin/${pname} $out/bin/hiddify
```
Comment (SuperSandro2000) [inline pkgs/by-name/hi/hiddify-next/package.nix:20]:
```suggestion
    mv $out/bin/hiddify-next $out/bin/hiddify
```

**Comment by linsui** [comment]
Since hidiffy violates the license used by sing-box I thought we shouldn't package it.

**Comment by wenjinnn** [comment]
Well, I'm ok with that reason. should we close this PR now or wait for confirmation from others?

**Comment by wenjinnn** [comment]
Clost this since #326367 pack it from source.

**Pull request closed without merging**